radix-transaction-scenarios/src/scenarios/access_controller_v2.rs

```diff
@@ -0,0 +1,154 @@
+use crate::internal_prelude::*;
+use radix_engine::updates::{ProtocolUpdate, ProtocolVersion};
+use radix_engine_interface::blueprints::access_controller::*;
+use radix_engine_interface::*;
+
+pub struct AccessControllerV2ScenarioConfig {
+    primary_role_private_key: PrivateKey,
+    recovery_role_private_key: PrivateKey,
+    confirmation_role_private_key: PrivateKey,
+}
+
+impl AccessControllerV2ScenarioConfig {
+    pub fn access_rule(&self, selector: impl FnOnce(&Self) -> &PrivateKey) -> AccessRule {
+        rule!(require(NonFungibleGlobalId::from_public_key(
+            &selector(self).public_key()
+        )))
+    }
+
+    pub fn rule_set(&self) -> RuleSet {
+        RuleSet {
+            primary_role: self.access_rule(|this| &this.primary_role_private_key),
+            recovery_role: self.access_rule(|this| &this.recovery_role_private_key),
+            confirmation_role: self.access_rule(|this| &this.confirmation_role_private_key),
+        }
+    }
+}
+
+#[derive(Default)]
+pub struct AccessControllerV2ScenarioState {
+    pub(crate) access_controller_component_address: State<ComponentAddress>,
+}
+
+impl Default for AccessControllerV2ScenarioConfig {
+    fn default() -> Self {
+        Self {
+            primary_role_private_key: Ed25519PrivateKey::from_u64(1).unwrap().into(),
+            recovery_role_private_key: Ed25519PrivateKey::from_u64(2).unwrap().into(),
+            confirmation_role_private_key: Ed25519PrivateKey::from_u64(3).unwrap().into(),
```

Comment:
I remember that we have some methods defining such keys. Maybe we can reuse that?

Reply:
Sounds good, will use that.